group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1867,11 +1936,12 @@ private Assignment updateTargetAssignment(
                     .withTopicsImage(metadataImage.topics())
                     .addOrUpdateMember(updatedMember.memberId(), updatedMember);
 
-            if (staticMemberReplaced) {
+            String previousMemberId = group.staticMemberId(updatedMember.instanceId());
```

Comment:
why do we do this vs. member.memberId?

Reply:
We check the member id associated to the instance id. If it is different from the member id of the current member, it means that the member was replaced. Using `member.memberId` does not work because it is the idea of the current member.

Reply:
Ok. I guess the final piece that was missing for me was where we set the updated member so that group.staticMemberId(updatedMember.instanceId()) returns the new memberID. This is probably just me not remembering the flow of the code though.

Reply:
Ah I see. we do it when we replay. The difference is that member is the new one and not the old one when we call this method.